lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralLinearModelAlgorithm.cxx

```diff
@@ -562,8 +567,16 @@ Scalar GeneralLinearModelAlgorithm::maximizeReducedLogLikelihood()
   OptimizationProblem problem(reducedLogLikelihoodFunction);
   problem.setMinimization(false);
   problem.setBounds(optimizationBounds_);
-  solver_.setStartingPoint(initialParameters);
   solver_.setProblem(problem);
+  try
+  {
+    // If the solver is single start, we can use its setStartingPoint method
+    solver_.setStartingPoint(initialParameters);
+  }
+  catch (NotYetImplementedException &) // setStartingPoint is not implemented in MultiStart
```

Comment:
It should be a NotDefinedException and the comment should be updated:
// setStartingPoint is not defined for the solver